src/main/java/frc/robot/subsystems/elevator/Elevator.java

```diff
@@ -0,0 +1,116 @@
+		this.elevatorStuff = elevatorStuff;
+		this.positionRequest = elevatorStuff.positionRequest();
+		this.voltageRequest = elevatorStuff.voltageRequest();
+
+		frontMotor.resetPosition(metersToMotorRotations(ElevatorConstants.MINIMUM_ACHIEVABLE_POSITION_METERS));
+		backMotor.resetPosition(metersToMotorRotations(ElevatorConstants.MINIMUM_ACHIEVABLE_POSITION_METERS));
+
+		this.commandBuilder = new ElevatorCommandsBuilder(this);
+
+		updateInputs();
+	}
+
+	public ElevatorCommandsBuilder getCommandBuilder() {
+		return commandBuilder;
+	}
+
+	protected void setPower(double power) {
+		frontMotor.setPower(power);
+		backMotor.setPower(power);
+	}
+
+	protected void setVoltage(double voltage) {
+		frontMotor.applyDoubleRequest(voltageRequest.withSetPoint(voltage));
+		backMotor.applyDoubleRequest(voltageRequest.withSetPoint(voltage));
+	}
+
+	protected void stop() {
+		frontMotor.stop();
+		backMotor.stop();
+	}
+
+	protected void setBrake(boolean brake) {
+		frontMotor.setBrake(brake);
+		backMotor.setBrake(brake);
+	}
+
+	protected void setTargetPosition(Rotation2d position) {
+		frontMotor.applyAngleRequest(positionRequest.withSetPoint(position));
+		backMotor.applyAngleRequest(positionRequest.withSetPoint(position));
+	}
+
+	protected boolean isAtBackwardLimit() {
+		return digitalInputsInputs.debouncedValue;
+	}
+
+	public double getPositionMeters() {
+		return motorRotationsToMeters(
+			Rotation2d.fromRotations(
+				(elevatorStuff.frontMotorStuff().positionSignal().getLatestValue().getRotations()
+					+ elevatorStuff.backMotorStuff().positionSignal().getLatestValue().getRotations()) / 2
+			)
+		);
+	}
+
+	protected void stayInPlace() {
+		frontMotor.applyAngleRequest(positionRequest.withSetPoint(elevatorStuff.frontMotorStuff().positionSignal().getLatestValue()));
+		backMotor.applyAngleRequest(positionRequest.withSetPoint(elevatorStuff.backMotorStuff().positionSignal().getLatestValue()));
+	}
+
+	public double motorRotationsToMeters(Rotation2d rotations) {
+		return rotations.getRotations() * elevatorStuff.motorRotationsToMetersConversionRatio();
+	}
+
+	public Rotation2d metersToMotorRotations(double meters) {
+		return Rotation2d.fromRotations(meters / elevatorStuff.motorRotationsToMetersConversionRatio());
+	}
+
+	protected void updateInputs() {
+		limitSwitch.updateInputs(digitalInputsInputs);
+		frontMotor.updateSignals(elevatorStuff.frontMotorStuff().positionSignal(), elevatorStuff.frontMotorStuff().voltageSignal());
+		backMotor.updateSignals(elevatorStuff.backMotorStuff().positionSignal(), elevatorStuff.backMotorStuff().voltageSignal());
+
+		Logger.processInputs(elevatorStuff.digitalInputsLogPath(), digitalInputsInputs);
+		Logger.recordOutput(getLogPath() + "isAtBackwardLimit", isAtBackwardLimit());
+		Logger.recordOutput(getLogPath() + "elevatorPosition", getPositionMeters());
+	}
+
+	@Override
+	protected void subsystemPeriodic() {
+		if (ElevatorConstants.MINIMUM_ACHIEVABLE_POSITION_METERS > getPositionMeters()) {
```

Comment:
consider doing with median filter, like in syncopa pivot

Reply:
I don't think it's a must because we don't need perfect accuracy here

Reply:
no need since from what I've seen the output looks continuous